ground-core/src/main/java/edu/berkeley/ground/db/CassandraClient.java

```diff
@@ -1,217 +1,202 @@
+    // hack to keep JGraphT up to date
+    if (table.equals("node_version")) {
+      long id = -1;
+      for (DbDataContainer container : insertValues) {
+        if (container.getField().equals("id")) {
+          id = (Long) container.getValue();
+        }
+      }
 
-    public CassandraConnection(Session session, DirectedGraph<Long, DefaultEdge> graph, PreparedStatement adjacencyStatement) {
-      this.session = session;
-      this.graph = graph;
-      this.adjacencyStatement = adjacencyStatement;
+      JGraphTUtils.addVertex(this.graph, id);
     }
+    if (table.equals("edge_version")) {
+      long nvFromId = -1;
+      long nvToId = -1;
 
-    /**
-     * Insert a new row into table with insertValues.
-     *
-     * @param table        the table to update
-     * @param insertValues the values to put into table
-     */
-    public void insert(String table, List<DbDataContainer> insertValues) {
-      // hack to keep JGraphT up to date
-      if (table.equals("node_version")) {
-        long id = -1;
-        for (DbDataContainer container : insertValues) {
-          if (container.getField().equals("id")) {
-            id = (Long) container.getValue();
-          }
+      for (DbDataContainer container : insertValues) {
+        if (container.getField().equals("from_node_version_id")) {
+          nvFromId = (Long) container.getValue();
         }
 
-        JGraphTUtils.addVertex(this.graph, id);
-      }
-      if (table.equals("edge_version")) {
-        long nvFromId = -1;
-        long nvToId = -1;
-
-        for (DbDataContainer container : insertValues) {
-          if (container.getField().equals("from_node_version_id")) {
-            nvFromId = (Long) container.getValue();
-          }
-
-          if (container.getField().equals("to_node_version_id")) {
-            nvToId = (Long) container.getValue();
-          }
+        if (container.getField().equals("to_node_version_id")) {
+          nvToId = (Long) container.getValue();
         }
-
-        JGraphTUtils.addEdge(this.graph, nvFromId, nvToId);
       }
 
-      String fieldsString = insertValues.stream()
-          .map(DbDataContainer::getField)
-          .collect(Collectors.joining(", "));
-      String actualValuesString = String.join(", ",
-          Collections.nCopies(insertValues.size(), "?"));
-
-      String insertString = "insert into " + table + "(" + fieldsString + ")";
-      String valuesString = "values (" + actualValuesString + ")";
-
-      String prepString = insertString + valuesString + ";";
-
-      BoundStatement statement = new BoundStatement(session.prepare(prepString));
+      JGraphTUtils.addEdge(this.graph, nvFromId, nvToId);
+    }
 
-      int index = 0;
-      for (DbDataContainer container : insertValues) {
-        CassandraClient.setValue(statement, container.getValue(), container.getGroundType(), index);
+    String fields =
+        insertValues.stream().map(DbDataContainer::getField).collect(Collectors.joining(", "));
```

Comment:
Do we know if there's overhead associated with converting `List`s to `Stream`s? I'm assuming they've optimized things enough for there not to be, but it'd be good to know for sure.